src/payment/checkout/payment-form/CardDetails.jsx

```diff
@@ -54,8 +54,157 @@ export class CardDetailsComponent extends React.Component {
     ];
   }
 
+  renderExperimentExpirationMonthOptions() {
+    const getMonthOptions = () => {
+      const pad = month => `0${month}`.slice(-2);
+
+      const items = [];
+      for (let i = 1; i <= 12; i += 1) {
+      // Display months in the MM format
+        items.push(<option key={i} value={i}>{pad(i)}</option>);
+      }
+      return items;
+    };
+
+    const monthText = this.props.intl.formatMessage(messages['payment.page.card.details.month']);
+
+    return [
+      <option key="Month" value="">{monthText}</option>,
+      ...getMonthOptions(),
+    ];
+  }
+
+  renderExperimentExpirationYearOptions() {
+    const getYearOptions = (startYear, endYear) => {
+      const items = [];
+      for (let i = startYear; i <= endYear; i += 1) {
+        // Display years in the YY format
+        items.push(<option key={i} value={i}>{i.toString().slice(-2)}</option>);
+      }
+      return items;
+    };
+
+    const yearText = this.props.intl.formatMessage(messages['payment.page.card.details.year']);
+    const currentYear = new Date().getFullYear();
+
+    return [
+      <option key="Year" value="">{yearText}</option>,
+      ...getYearOptions(currentYear, currentYear + 15),
```

Comment:
is +15 a standard amount?

Reply:
1. Yes, tested both of those locally, but I think stage will be more useful for getting a more accurate idea of whether this really works
2. I'm lazy and use the `Squash and merge` button. The PR has the ticket link in the top level comment
3. Shelby's been out for a while, but maybe @emma-green can fill in. I believe the goal was to try and reach significance on revenue, but I agree that seems unlikely. If we want to try to add a bigger change, I could add card masking to this PR. Emma was thinking that card masking  might not need to be an experiment, but could just be added separately

Reply:
Oh and the `+15` is from https://github.com/edx/frontend-app-payment/blob/58a0aa018f4741118d2342757c6c2d0d4c1a05ac/src/payment/checkout/payment-form/CardDetails.jsx#L47-L55

Not sure why `+15`, but it seems pretty reasonable to me

Reply:
1. can't you just test against stage using https://local.stage.edx.org:1998?
3. i think the card mask goes into the category of good UX features that probably won't shift revenue, so i doubt that would address the issue of the sum total changes being potentially too small. Not necessarily a blocker to merging, but something we should have an answer to before starting the experiment.

Reply:
1. The local run stage stuff has never worked for me unfortunately
2. I'll let Emma and product decide. If this stuff doesn't give a significant impact though then maybe we either shouldn't do it or just add it anyway if we're really attached

Reply:
1. ah interesting, i've only ever used that locally actually, but sure i guess testing on stage could work then
2. yup their decision, though my take on it would probably be that we could either add even more changes/add back some of the scrapped changes, consider just implementing these changes without a test or consider implementing them with a test even if it doesn't reach significance for 5% or over